B 19 add filters for pages by role

Rooman opened the pull request.

OleksandrGubariev committed:
Add Filter

OleksandrGubariev committed:
Add Filter

Commit:
[b-19] Implement security

Brutt commented on pom.xml:
```diff
@@ -75,11 +75,7 @@
-            <artifactId>thymeleaf-extras-java8time</artifactId>
-            <version>3.0.4.RELEASE</version>
-        </dependency>
+
```
exclude pom.xml from PR

Brutt commented on src/main/java/com/photostudio/web/filter/SecurityFilter.java:
```diff
@@ -0,0 +1,105 @@
+import static com.photostudio.entity.user.UserRole.*;
+
+
+@Slf4j
```
Use approach as implemented in other classes

Rooman replied:
Не менять.

Brutt commented on src/main/java/com/photostudio/web/filter/SecurityFilter.java:
```diff
@@ -0,0 +1,105 @@
+        urlToRoleMap.put("/logout", USER);
+        urlToRoleMap.put("/orders", USER);
+        // Optional, included by default
+        urlToRoleMap.put("/admin/order", ADMIN);
```
we don`t have this endpoint

Brutt commented on src/main/java/com/photostudio/web/filter/SecurityFilter.java:
```diff
@@ -0,0 +1,105 @@
+            if (hasAccess(requiredRole, currentUser.getUserRole())) {
+                chain.doFilter(request, response);
+            } else {
+                response.sendRedirect(request.getContextPath() + "/fuck_you_asshole");
```
change endpoint name

Brutt commented on src/main/java/com/photostudio/web/templater/TemplateEngineFactory.java:
```diff
@@ -33,7 +31,7 @@ public static void configTemplate(ServletContext servletContext) {
 
     public static void process(String template, Map<String, Object> productsMap, Writer writer) {
         IContext context = new Context(Locale.getDefault(), productsMap);
-        TEMPLATE_ENGINE.addDialect(new Java8TimeDialect());
```
do not overwrite develop`s branch changes

Yuliia-Sh commented on src/main/resources/application.properties:
```diff
@@ -1,5 +1,5 @@
 jdbc.url=jdbc:mysql://localhost:3306/photostudio
 jdbc.user=root
-jdbc.password=toor
+jdbc.password=root
```
password for server will  be incorrect, won't it?

Brutt replied:
Server doesn't use credentials from property file. It uses credentials from system variables

OleksandrGubariev committed:
1.Change endpoint name in class SecurityFilter
2.Change pom.xml - add thymeleaf dependency
3.Change TemplateEngineFactory - add Java8TimeDialect

OleksandrGubariev committed:
1.Change endpoint name in SecurityFilter
2.delete application.properties

OleksandrGubariev committed:
add application.properties

OleksandrGubariev committed:
change application.properties
